ecosystem/opUSDC/op-usdc.md

```diff
@@ -0,0 +1,113 @@
+
+To address this pain point, Circle has rolled out the [Bridged USDC Standard](https://www.circle.com/blog/bridged-usdc-standard) to ensure that chains can easily migrate into the Native USDC version of the token in the future and prevent the fragmentation problem.
+
+However, the Bridged USDC Standard as defined does not work with the current OP Stack `StandardBridge` specifications, causing confusion from chain operators on how to conveniently host USDC on their chain. Circle’s primary goal is to enable chain developers to easily deploy the Bridged USDC Standard. 
+
+To address the issues faced by chain operators and align with Circle's motivations, OpUSDC is proposed as a reference implementation for OP Stack chain operators. This allows them to comply with the Bridged USDC standard and potentially access native USDC.
+
+# Alternatives Considered
+
+**Do nothing**
+
+The `StandardBridge` contracts have been implemented since the chain's inception, allowing the use of any common ERC20 token, such as USDC. As the number of OP Chains within the Superchain ecosystem grows, a problematic scenario will appear where an OP Chain might host different implementations of the same asset: *legacy* bridged USDC, new bridged USDC that accomplish with the Superc20, 3rd party bridged USDC, and native USDC. This fragmentation would significantly complicate the migration to native USDC and negatively impact the user experience.
+
+**Dedicated shared bridge**
+
+To align architecturally with the Superchain design, a proposal was made to enable full fungibility of opUSDC across OP Chains within the same interoperability cluster. This design featured a `SharedL1Adapter` and a factory that allowed trustless deployments of a fungible `bridgedUSDC` implementation on any chain. Additionally, it included access control restrictions for contract upgrades and migrations to native USDC. However, this design required a trusted third party to manage the relevant keys, introducing various layers of liability. These concerns did not meet the issuer's requirements at the time but may be revisited in the future.
+
+# Proposed Solution
+
+The solution involves a [Custom Token Bridge](https://docs.optimism.io/builders/app-developers/bridging/custom-bridge) implementation, which provides desired features such as pausability and the ability to perform migrations. The OP Stack security model secures the proposed design, similar to how `StandardBridge` operates today.
+
+### Implementation
+
+There are four main pieces:
+
+- `BridgedUSDC`: An upgradable contract that implements the Bridged USDC token standard.
+- `L1OpUSDCFactory`: A factory on L1 that implements cross-chain deployment logic.
+- `L1OpUSDCBridgeAdapter`: An upgradeable bridge contract that locks and secures USDC. It is also responsible for triggering the migration and properly burning the USDC amount once it is completed.
```

Comment:
Hey! To clarify the "Bridged USDC" token standard is not a standard for the actual contract code, but a standard for conforming to circle's processes should a chain operator ever want to migrate the Bridged USDC to Native USDC [ref](https://github.com/circlefin/stablecoin-evm/blob/master/doc/bridged_USDC_standard.md).

Not sure I fully understand the other question, but USDC has a built in minter access control that the chain operator would be able to use if they ever wanted to add another third-party bridge as a minter as an extra route.

Reply:
Thanks for your response! We are familiar with Circle's standard and implemented the support of it in this [PR](https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3618).

However, if lets say a chain operator does want to allow a Hyperlane message to mint the USDC, presumably they also want to allow redemption of the collateral the other way round in `L1OpUSDCBridgeAdapter` which doesn't have minter access control? That would have to be built into `L1OpUSDCBridgeAdapter`?

Reply:
Sorry let me be more explicit. Let's say a chain operator starts with the "vanilla" setup of having the `L2OpUSDCBridgeAdapter` for minting and burning the L2 USDC. $1M USDC has been deposited. To avoid the L2 withdrawal times, they would like Hyperlane (or anybody to be able to be used to withdraw. The Hyperlane adapter can be added as a minter on the `FiatToken` contract on L2, but it would need to be able to release the collateral on the `L1OpUSDCBridgeAdapter` to the user withdrawing. Does that make sense? Of course you could require that deposits have to be on the Hyperlane adapter, but the fragmentation of the liquidity seems not ideal. IMO `L1OpUSDCBridgeAdapter` easily can fulfill the `Lockbox` responsabilities in xERC20 terms.

Good to know that chain operators could upgrade the contract to add this logic, but I think for all parties involved it seems safer if this logic was on the default implementation?

Reply:
We don't really want to incentivize that type of upgrades or modifications. Ideally end-users will be able to still use across or any other fast transport layer. That would in-turn use the bridgedUSDC standard as the underlying asset.

The main goal here is to be compliant with with circle's bridged standard out of the box to make migration to native as seemless as possible, allowing additional minters is already "non-standard" for their process [as seen here in point 1](https://github.com/circlefin/stablecoin-evm/blob/master/doc/bridged_USDC_standard.md#bridge-contracts)

"Ability to pause USDC bridging to create a lock on the supply."

With multiple minters this already breaks this functionality as there is a bunch of extra off-chain coordination that needs to happen to create a supply lock. The main goal here is to be able to fast-track different op-stack chains to be able to migrate to native, to expand native usdc liquidity across the superchain.

We understand that Native USDC may not be a priority for some chains, which is why we allow for the chain operator to ultimately decide what to do.

Reply:
I'm biased obviously, but IMO by not making this type of modification accessible, chain operators will have to choose between choosing a different incompatible token standard to support fast-withdrawals or this. Ultimately, I respect your decision and appreciate you engaging with me on this discussion.